141-linked-list-cycle.py

```diff
@@ -6,7 +6,27 @@
 
 class Solution:
   def hasCycle(self, head: ListNode) -> bool:
-    return detect_cycle_bruteforce(head)
+    return detect_cycle(head)
+
+def detect_cycle(head):
+  if not head:
+    return False
+
+  slow = head
+  fast = slow.next
+
+  while slow and fast:
+    if slow == fast:
+      return True
+
+    slow = slow.next
+    if fast.next and fast.next.next:
+      fast = fast.next.next
+    else:
+      fast = None
+
+  return False
```

Comment:
Disregard this. I messed up git's history a bit and don't really want to unravel it :)

Reply:
Okay 😄